From 16c216d5dc686037686bb48f46708f687c58ae42 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Horia=20Geant=C4=83?= <horia.geanta@nxp.com>
Date: Thu, 14 Mar 2019 14:46:48 +0200
Subject: [PATCH 101/235] crypto: caam - fix compilation warning
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Fix the following compilation warning:

drivers/crypto/caam/caamalg.c: In function 'caam_algapi_init':
drivers/crypto/caam/caamalg.c:3707:19: warning: 't_alg' may be used uninitialized in this function [-Wmaybe-uninitialized]
       (t_alg->caam.class1_alg_type & OP_ALG_AAI_MASK) ==
                   ^

Fixes: d7ddff4499e8 ("crypto: caam - add register map changes cf. Era 10")
Signed-off-by: Horia GeantÄƒ <horia.geanta@nxp.com>
[Xulin: Original patch taken from NXP LSDK-19.06.]
Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 drivers/crypto/caam/caamalg.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/crypto/caam/caamalg.c b/drivers/crypto/caam/caamalg.c
index a9cc90f98277..0b58614378cd 100644
--- a/drivers/crypto/caam/caamalg.c
+++ b/drivers/crypto/caam/caamalg.c
@@ -3708,7 +3708,7 @@ int caam_algapi_init(struct device *ctrldev)
 		 * on LP devices.
 		 */
 		if (aes_vid == CHA_VER_VID_AES_LP &&
-		    (t_alg->caam.class1_alg_type & OP_ALG_AAI_MASK) ==
+		    (alg->class1_alg_type & OP_ALG_AAI_MASK) ==
 		    OP_ALG_AAI_XTS)
 			continue;
 
-- 
2.17.1

